MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,344 @@
+# SPA-JS v2 Migration Guide
+
+With the v2 release of Auth0-SPA-JS, we have improved both performance and developer experience while trying to limit the amount of breaking changes. However, as with any major version bump, v2 of Auth0-SPA-JS contains a set of breaking changes.
+
+**Please review this guide thoroughly to understand the changes required to migrate your application to v2.**
+
+- [Polyfills and supported browsers](#polyfills-and-supported-browsers)
+- [Public API changes](#polyfills-and-supported-browsers)
+  - [client_id has been renamed to clientId](#client_id-has-been-renamed-to-clientid)
+  - [Introduction of authorizationParams](#introduction-of-authorizationparams)
+  - [Introduction of logoutParams](#introduction-of-logoutparameters)
+  - [buildAuthorizeUrl has been removed](#buildauthorizeurl-has-been-removed)
+  - [buildLogoutUrl has been removed](#buildlogouturl-has-been-removed)
+  - [redirectMode has been removed from loginWithRedirect](#redirectmode-has-been-removed-from-loginwithredirect)
+  - [ignoreCache on getTokenSilentlyhas been replaced by cacheMode](#ignorecache-on-gettokensilentlyhas-been-replaced-by-cachemode)
+  - [application/x-www-form-urlencoded is used by default instead of application/json](#applicationx-www-form-urlencoded-is-used-by-default-instead-of-applicationjson)
+  - [No more iframe fallback by default when using refresh tokens](#no-more-iframe-fallback-by-default-when-using-refresh-tokens)
+  - [getUser and getIdTokenClaims](#getuser-and-getidtokenclaims)
+  - [Changes to default scopes (profile and email)](#changes-to-default-scopes-profile-and-email)
+    - [defaultScope](#defaultscope)
+  - [checkSession no longer throws exceptions](#checksession-no-longer-throws-exceptions)
+  - [getIdTokenClaimsOptions type has been removed](#getidtokenclaimsoptions-type-has-been-removed)
+- [Module output](#module-output)
+  - [No more default export](#no-more-default-export)
+  - [Changes on how to create an instance when relying on globally available API’s (e.g. using CDN)](#changes-on-how-to-create-an-instance-when-relying-on-globally-available-apis-eg-using-cdn)
+
+## Polyfills and supported browsers
+
+As [Microsoft has dropped support for IE11](https://blogs.windows.com/windowsexperience/2022/06/15/internet-explorer-11-has-retired-and-is-officially-out-of-support-what-you-need-to-know) themselves, as of v2 our SDK <u>no longer supports IE11</u>. Because of that, we have dropped including any polyfill in our bundle, as all of these polyfills were for IE11. On top of that, our bundle output is now set to ES2017, which should work fine on all major browsers.
```

Comment:
I would reword bits of this:

```suggestion
As [Microsoft has dropped support for IE11](https://blogs.windows.com/windowsexperience/2022/06/15/internet-explorer-11-has-retired-and-is-officially-out-of-support-what-you-need-to-know), we are no longer including any polyfills in our bundle, as all of these polyfills were for IE11. Therefore <u>we no longer support IE11 from v2 of this SDK</u>. In addition, our bundle output is now set to ES2017, which should work fine on all major browsers.
```

Reply:
Additionally, will we provide a list anywhere of the polyfills that should be used if the app developer wants to include them themselves?

Reply:
We have elaborated on that in the PR description, which will be linked from the changelog: https://github.com/auth0/auth0-spa-js/pull/951. 

Maybe it makes sense to add every PR to the corresponding section in the migration guide, but I don't want to cause to much confusion by pulling people to PRs if they don't need to.